client/components/prompt-input/PromptInputForm.tsx

```diff
@@ -111,36 +123,49 @@ const PromptInputForm = () => {
       >
         Start your learning journey here:
       </Typography>
-      <Box
-        width="700px"
-        display="flex"
-        flexDirection="column"
-        gap={2}
-      >
-        <UserInputBar
-          type="text"
-          value={inputValue}
-          onChange={handleInputChange}
-        />
-        <FormControlLabel
-          control={
-            <Checkbox
-              checked={isChecked}
-              onChange={handleCheckboxClick}
-              color="primary"
-            />
-          }
-          label="Content on this site may be generated by artifical intelligence (AI). Please use discretion and verify information where necessary."
-          sx={{ color: 'text.primary' }}
-        />
-        <GenerateButton
-          disabled={!isChecked}
-          sx={{ alignSelf: 'center' }}
-          onClick={handleSubmit}
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Box
+          width="700px"
+          display="flex"
+          flexDirection="column"
+          gap={2}
         >
-          Generate
-        </GenerateButton>
-      </Box>
+          <Controller
+            name="input"
+            control={control}
+            render={({ field }) => (
+              <UserInputBar
+                {...field}
+                error={!!errors.input}
+                helperText={errors.input ? errors.input.message : ''}
+              />
+            )}
+          />
+          <FormControlLabel
+            control={
+              <Controller
+                name="isChecked"
+                control={control}
+                render={({ field }) => (
+                  <UserInputCheckbox
+                    {...field}
+                    checked={field.value}
+                  />
+                )}
+              />
+            }
+            label="Content on this site may be generated by artifical intelligence (AI). Please use discretion and verify information where necessary."
+            sx={{ color: 'text.primary' }}
+          />
+          <GenerateButton
```

Comment:
Are handling loading states in the UI part of this ticket?
Also it looks like the Figma has a gray color for the checkbox rather thant the green seen here. Not sure if this part of this PR.

Reply:
The Loading/Submitting states are part of a separate ticket that I'm in the process of implementing. As for the checkbox, I must've set the color incorrectly, although it was never faithful to the Figma design to begin with. I am planning on assigning myself a ticket to style the UserInput UI the way it's presented in the updated Figma design.

Reply:
👍